src/components/wms/InformationPanel.vue

```diff
@@ -0,0 +1,204 @@
+      </template>
+      <v-list>
+        <v-list-item
+          :title="props.layerTitle"
+          :subtitle="analysisTime"
+          :prepend-icon="layersIcon"
+        >
+        </v-list-item>
+        <v-list-item
+          :title="'Time range'"
+          :subtitle="formattedTimeRange"
+          :prepend-icon="timeIcon"
+        >
+        </v-list-item>
+        <v-list-group>
+          <template v-slot:activator="{ props }">
+            <v-list-item
+              v-bind="props"
+              title="Color scales"
+              :prepend-icon="colorScalesIcon"
+            ></v-list-item>
+          </template>
+          <v-list-item
+            v-for="(style, index) in props.styles"
+            :key="index"
+            :title="style.title"
+            :subtitle="style.name"
+            @click="onStyleClick(style)"
+            :class="{ selected: isSelected(style) }"
+          >
+          </v-list-item>
+          <v-list-item v-if="mutableColorScaleRange" :prepend-icon="rangeIcon">
+            <v-row align="center">
+              <v-col cols="6">
+                <v-text-field
+                  v-model.number="mutableColorScaleRange.min"
+                  label="Min"
+                  outlined
+                  clearable
+                  :rules="[
+                    rules.required,
+                    rules.biggerThanZero,
+                    rules.smallerThanMax,
+                  ]"
+                ></v-text-field>
+              </v-col>
+              <v-col cols="6">
+                <v-text-field
+                  v-model.number="mutableColorScaleRange.max"
+                  label="Max"
+                  outlined
+                  clearable
+                  :rules="[rules.required, rules.biggerThanMin]"
+                ></v-text-field>
+              </v-col>
+            </v-row>
+            <v-btn
+              color="primary"
+              variant="text"
+              @click="changecolorScaleRange"
+            >
+              Submit
+            </v-btn>
+          </v-list-item>
+        </v-list-group>
+        <v-list-item v-if="props.completelyMissing">
+          Wms layer is completely missing
+        </v-list-item>
+      </v-list>
+    </v-menu>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue'
+import { DateTime } from 'luxon'
+import { Style } from '@deltares/fews-wms-requests'
+import { ref } from 'vue'
+
+interface Props {
+  layerTitle: string
+  currentTime: Date | null
+  forecastTime: Date | null
+  styles: Style[] | null
+  completelyMissing: boolean | null
+  firstValueTime: Date | null
+  lastValueTime: Date | null
+  colorScaleRange: { min: number; max: number } | null | undefined
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  layerTitle: '',
+  currentTime: null,
+  forecastTime: null,
+  styles: null,
+  completelyMissing: null,
+  firstValueTime: null,
+  lastValueTime: null,
+  colorScaleRange: null,
```

Comment:
I would make most of these like `currentTime?: Date` such that undefined is the missing data value.
Then we don't have to give defaults below for these and we can just pass values in the rest of the code instead of doing:
```ts
      :completelyMissing="props.layerCapabilities?.completelyMissing ?? null"
```
We can then do:
```ts
      :completelyMissing="props.layerCapabilities?.completelyMissing"
```

This is for most of the places using `null`.

Reply:
this is fine, except for some of the `props.layerCapabilities` that are sometimes defined as `completelyMissing?: boolean | null` within the `fews-wms-requests`